node/p2p_block_sync.md

```diff
@@ -0,0 +1,84 @@
+# P2P Block Sync
+
+## Abstract
+
+P2P Block Sync enables rollkit full nodes, including aggregators, to gossip blocks amongst  
+themselves and have rollup lower block times than the DA layer block times.
+
+```mermaid
+sequenceDiagram
+    title P2P Block Sync
+
+    participant User
+    participant Block Producer
+    participant Full Node 1
+    participant Full Node 2
+    participant DA Layer
+
+    User->>Block Producer: Send Tx
+    Block Producer->>Block Producer: Generate Block
+    Block Producer->>DA Layer: Publish Block
+
+    Block Producer->>Full Node 1: Gossip Block
+    Block Producer->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Verify Block
+    Full Node 1->>Full Node 2: Gossip Block
+    Full Node 1->>Full Node 1: Mark Block Soft-Confirmed
+
+    Full Node 2->>Full Node 2: Verify Block
+    Full Node 2->>Full Node 2: Mark Block Soft-Confirmed
+
+    DA Layer->>Full Node 1: Retrieve Block
+    Full Node 1->>Full Node 1: Mark Block Hard-Confirmed
+
+    DA Layer->>Full Node 2: Retrieve Block
+    Full Node 2->>Full Node 2: Mark Block Hard-Confirmed
+```
+
+## Protocol/Component Description
+
+P2P Block Sync consists of the following components:
+
+* block exchange service: responsible for gossiping blocks over P2P
+* block publication to P2P network
+* block retrieval from P2P network
+
+### Block Exchange Service
+
+The block exchange service is created during full node initialization. After that, during the block manager's initialization, a pointer to the block store inside block exchange service is passed to it. Blocks created in the block manager which need to gossipped over a p2p network are then passed to the `BlockCh` channel and then sent to [go-header] service to be gossip blocks over P2P network.
+
+### Block Publication to P2P network
+
+Blocks ready to be published to the P2P network are sent to the `BlockCh` channel in Block Manager inside `publishLoop`.
```

Comment:
I would differentiate the 2 paths of gossiping a block that i received either from the network or that i created myself. 

If I create a block myself do i write it into the block -store > put it in the channel and then try to write it into the store again after pulling it from the `blockPublishLoop`?

Reply:
Makes sense. Added differentiation. Only the block creator (sequencer) will be putting it in the channel which gets broadcasted by the P2P network. The validating full nodes just receive it in their block stores, read it, and apply it to update their state.